hedera-node/src/main/java/com/hedera/services/ledger/MerkleAccountScopedCheck.java

```diff
@@ -0,0 +1,100 @@
+package com.hedera.services.ledger;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.ledger.properties.AccountProperty;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.txns.validation.OptionValidator;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+
+import java.util.Map;
+
+import static com.hedera.services.ledger.properties.AccountProperty.BALANCE;
+import static com.hedera.services.ledger.properties.AccountProperty.EXPIRY;
+import static com.hedera.services.ledger.properties.AccountProperty.IS_DELETED;
+import static com.hedera.services.ledger.properties.AccountProperty.IS_SMART_CONTRACT;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
+
+public class MerkleAccountScopedCheck implements LedgerCheck<MerkleAccount, AccountProperty> {
+
+	private final GlobalDynamicProperties dynamicProperties;
+	private final OptionValidator validator;
+	private BalanceChange balanceChange;
+
+	MerkleAccountScopedCheck(
+			GlobalDynamicProperties dynamicProperties,
+			OptionValidator validator) {
+		this.dynamicProperties = dynamicProperties;
+		this.validator = validator;
+	}
+
+	@Override
+	public ResponseCodeEnum checkUsing(final MerkleAccount account, final Map<AccountProperty, Object> changeSet) {
```

Comment:
nit: this method not only does checking, it also does updating. Maybe rename it to something like `checkAndUpdateUsing`?

Reply:
good point, but as this is a MerkleAccount check, we are updating the balance. But if the type of entity changes I dont think we will have a balance aspect of the entity to update. In that case I dont want to rename the Interface method.

Reply:
Understood.